src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_exc_grad.hpp

```diff
@@ -134,6 +135,21 @@ void ReferenceReplicatedXCHostIntegrator<ValueType>::
       host_data.vgamma     .resize( npts );
     }
 
+    if( func.is_mgga() ) {
+      host_data.basis_eval .resize( 10 * npts * nbe );
```

Comment:
I know I'm not the best at this either - but it might be good to comment why these factors are here. Magic numbers are hard to maintain, especially since they often contain contributions from multiple places (e.g. derivatives, spin, etc)